WooCommerce/src/main/kotlin/com/woocommerce/android/ui/bookings/BookingMapper.kt

```diff
@@ -100,14 +100,23 @@ class BookingMapper @Inject constructor(
         )
     }
 
-    private fun BookingEntity.Status.toUiModel(): BookingStatus = when (this) {
-        BookingEntity.Status.Paid -> BookingStatus.Paid
-        BookingEntity.Status.PendingConfirmation -> BookingStatus.PendingConfirmation
-        BookingEntity.Status.Cancelled -> BookingStatus.Cancelled
-        BookingEntity.Status.Complete -> BookingStatus.Complete
-        BookingEntity.Status.Confirmed -> BookingStatus.Confirmed
-        BookingEntity.Status.Unpaid -> BookingStatus.Unpaid
-        is BookingEntity.Status.Unknown -> BookingStatus.Unknown(this.key)
+    private fun BookingEntity.Status.toUiModel(
+        orderStatus: String?,
+        paymentMethod: String?,
+    ): BookingStatus {
+        return if (orderStatus == "completed" && paymentMethod == "cod") {
```

Comment:
I believe the condition should `!= "completed"` to match the web logic:


```suggestion
        return if (orderStatus != "completed" && paymentMethod == "cod") {
```

Reply:
🤦 yes, let me fix that.

Reply:
Fixed